lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -633,7 +700,8 @@ SILFunction *PromotedParamCloner::initCloned(SILOptFunctionBuilder &FuncBuilder,
          && "SILFunction missing DebugScope");
   assert(!Orig->isGlobalInit() && "Global initializer cannot be cloned");
   auto *Fn = FuncBuilder.createFunction(
-      SILLinkage::Shared, ClonedName, ClonedTy, Orig->getGenericEnvironment(),
+      swift::getSpecializedLinkage(Orig, Orig->getLinkage()),
```

Comment:
I talked with @meg-gupta. I think this is the correct thing to do here. Can you just add a comment in the commit why you are doing this? Namely that we are matching what other code does and that this handles private correctly since private -> shared doesn't make sense.